packages/dashboard-core-plugins/src/redux/actions.ts

```diff
@@ -9,14 +9,28 @@ import { FilterSet } from '../panels';
 import { Link } from '../linker/LinkerUtils';
 import { ColumnSelectionValidator } from '../linker/ColumnSelectionValidator';
 
+export interface SessionConfig {
+  type: string;
+  id: string;
+}
+
 export interface SessionWrapper {
   session: IdeSession;
   connection: IdeConnection;
-  config: {
-    type: string;
-    id: string;
-  };
+  config: SessionConfig;
 }
+
+/**
+ * Set the connection for the dashboard specified
+ * @param id The ID of the dashboard to set the connection for
+ * @param connection The connection object to set for the dashboard
+ */
+export const setDashboardConnection = (
```

Comment:
Are we planning to use separate connections per dashboard at some point?

Reply:
No plan yet.

Reply:
Just wondering why we store connections per dashboard, not just a global one.

Reply:
Didn't think there was a need to limit it to a global connection when could make it configurable so each dashboard has it's own.